harmonica/tests/test_transformations.py

```diff
@@ -139,3 +235,159 @@ def test_derivative_upward_order2(sample_potential, sample_g_zz):
+    # need to drop upward coordinate (bug in xrft)
+    potential_padded = xrft.pad(
+        sample_potential.drop_vars("upward"),
+        pad_width=pad_width,
+    )
+    # Calculate northing derivative and unpad it
+    derivative = derivative_northing(potential_padded)
+    derivative = xrft.unpad(derivative, pad_width)
+    # Compare against g_n (trim the borders to ignore boundary effects)
+    trim = 6
+    derivative = derivative[trim:-trim, trim:-trim]
+    g_n = sample_g_n[trim:-trim, trim:-trim] * 1e-5  # convert to SI units
+    rms = root_mean_square_error(derivative, g_n)
+    assert rms / np.abs(g_n).max() < 0.1
+
+
+def test_derivative_northing_order2(sample_potential, sample_g_nn):
+    """
+    Test higher order of derivative_northing function against the sample grid
+    """
+    # Pad the potential field grid to improve accuracy
+    pad_width = {
+        "easting": sample_potential.easting.size // 3,
+        "northing": sample_potential.northing.size // 3,
+    }
+    # need to drop upward coordinate (bug in xrft)
+    potential_padded = xrft.pad(
+        sample_potential.drop_vars("upward"),
+        pad_width=pad_width,
+    )
+    # Calculate second northing derivative and unpad it
+    second_deriv = derivative_northing(potential_padded, order=2)
+    second_deriv = xrft.unpad(second_deriv, pad_width)
+    # Compare against g_nn (trim the borders to ignore boundary effects)
+    trim = 6
+    second_deriv = second_deriv[trim:-trim, trim:-trim]
+    g_nn = sample_g_nn[trim:-trim, trim:-trim] * 1e-9  # convert to SI units
+    rms = root_mean_square_error(second_deriv, g_nn)
+    assert rms / np.abs(g_nn).max() < 0.1
+
+
+def test_laplace(sample_potential):
+    """
+    Test second order of derivative fullfill laplace equation
+    """
+    # Pad the potential field grid to improve accuracy
+    pad_width = {
+        "easting": sample_potential.easting.size // 3,
+        "northing": sample_potential.northing.size // 3,
+    }
+    # need to drop upward coordinate (bug in xrft)
+    potential_padded = xrft.pad(
+        sample_potential.drop_vars("upward"),
+        pad_width=pad_width,
+    )
+    # Calculate second northing derivative and unpad it
+    second_deriv_zz = derivative_upward(potential_padded, order=2)
+    second_deriv_ee = derivative_easting(potential_padded, order=2)
+    second_deriv_nn = derivative_northing(potential_padded, order=2)
+    second_deriv_zz = xrft.unpad(second_deriv_zz, pad_width)
+    second_deriv_ee = xrft.unpad(second_deriv_ee, pad_width)
+    second_deriv_nn = xrft.unpad(second_deriv_nn, pad_width)
+    # Compare g_nn + g_ee against -g_zz (trim the borders to ignore boundary
+    # effects)
+    trim = 6
+    second_deriv_sum = second_deriv_ee + second_deriv_nn
+    second_deriv_sum = second_deriv_sum[trim:-trim, trim:-trim]
+    second_deriv_zz = second_deriv_zz[trim:-trim, trim:-trim]
+    rms = root_mean_square_error(second_deriv_sum, -second_deriv_zz)
+    assert rms < 1e-20
```

Comment:
I've been using RMS on all the other tests mainly because the outputs of the FFT filters are not exactly the same as the ones computes through forward modelling. So the RMS offers a good compromise of accounting for most of the points, ignoring if some subset of them are actually not that similar (this happens specially in the boundaries).

But for this test I suspect that we don't need to use RMS, maybe we could just use `npt.assert_allclose()` and tweak the `atol` and/or the `rtol` if needed. I usually plot both to set a floor of error (`atol`) and try to keep the relative error to the lowest possible.

Reply:
since the data object is xarray, I change rms to `xarray.testing.assert_allclose()`

Reply:
Much better 👍🏼 !